third_party/terraform/tests/resource_dialogflow_intent_test.go.erb

```diff
@@ -0,0 +1,192 @@
+		project = google_project_service.agent_project.project
+		role    = "roles/dialogflow.admin"
+		member  = "serviceAccount:service-${google_project.agent_project.number}@gcp-sa-dialogflow.iam.gserviceaccount.com"
+		depends_on = [google_project_service.agent_project]
+	  }
+
+	resource "google_dialogflow_agent" "agent" {
+		project = "%s"
+		display_name = "%s"
+		default_language_code = "en"
+		time_zone = "America/New_York"
+		depends_on = [google_project_iam_member.agent_create]
+	  }
+	resource "google_dialogflow_intent" "foobar" {
+		depends_on = [google_dialogflow_agent.agent]
+		project = "%s"
+		display_name = "%s"
+	  }
+	`, projectID, projectID, orgID, projectID, agentName, projectID, intentName)
+}
+
+func testAccDialogflowIntent_full1(projectID string, orgID string, agentName string, intentName string) string {
+	return fmt.Sprintf(`
+	resource "google_project" "agent_project" {
+		project_id = "%s"
+		name       = "%s"
+		org_id     = "%s"
+	  }
+
+	resource "google_project_service" "agent_project" {
+		project = google_project.agent_project.project_id
+		service = "dialogflow.googleapis.com"
+	  }
+	  
+	resource "google_project_iam_member" "agent_create" {
+		project = google_project_service.agent_project.project
+		role    = "roles/dialogflow.admin"
+		member  = "serviceAccount:service-${google_project.agent_project.number}@gcp-sa-dialogflow.iam.gserviceaccount.com"
+		depends_on = [google_project_service.agent_project]
+	  }
+
+	resource "google_dialogflow_agent" "agent" {
+		project = "%s"
+		display_name = "%s"
+		default_language_code = "en"
+		time_zone = "America/New_York"
+		depends_on = [google_project_iam_member.agent_create]
+	  }
+
+	resource "google_dialogflow_intent" "foobar" {
+		depends_on = [google_dialogflow_agent.agent]
+		project = "%s"
+		display_name = "%s"
+		webhook_state = "WEBHOOK_STATE_ENABLED"
+		priority = 1
+		is_fallback = false
+		ml_disabled = true
+		action = "some_action"
+		reset_contexts = true
+		input_context_names = ["projects/%s/agent/sessions/-/contexts/some_id"]
+		events = ["some_event"]
+		default_response_platforms = ["FACEBOOK","SLACK"]
+	  }
+	`, projectID, projectID, orgID, projectID, agentName, projectID, intentName, projectID)
+}
+
+func testAccDialogflowIntent_full2(projectID string, orgID string, agentName string, intentName string) string {
+	return fmt.Sprintf(`
+	resource "google_project" "agent_project" {
+		project_id = "%s"
+		name       = "%s"
+		org_id     = "%s"
+	  }
+
+	resource "google_project_service" "agent_project" {
+		project = google_project.agent_project.project_id
+		service = "dialogflow.googleapis.com"
+	  }
+	  
+	resource "google_project_iam_member" "agent_create" {
+		project = google_project_service.agent_project.project
+		role    = "roles/dialogflow.admin"
+		member  = "serviceAccount:service-${google_project.agent_project.number}@gcp-sa-dialogflow.iam.gserviceaccount.com"
+		depends_on = [google_project_service.agent_project]
```

Comment:
This isn't needed because when you interpolate a value anywhere in this resource from an upstream resource (eg `project = google_project_service.agent_project.project`) the dependency is added implicitly.